packages/block-editor/src/components/writing-flow/use-multi-selection.js

```diff
@@ -87,6 +87,10 @@ export default function useMultiSelection() {
 	 * select the entire block contents.
 	 */
 	useEffect( () => {
```

Comment:
Maybe we should use `useRefEffect` here instead of `useEffect`?

Both `useSelectAll` and `useArrowNav` are using `useRefEffect` hook.

https://github.com/WordPress/gutenberg/blob/9e2fc57fe27dfca594bdff358853a0a0acaf8efe/packages/block-editor/src/components/writing-flow/index.js#L35-L37

P.S. I've got a chance to use this hook in action yet, but this seems like a good use case.

Reply:
I updated to use useRefEffect and indeed, it seems to work.